sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -500,46 +501,58 @@ export class StudentService extends RecordDataModelService<Student> {
   }
 
   /**
-   * Use the information available in the authentication token to update
-   * the user and student data currently on DB.
-   * @param studentToken student authentication token.
+   * Updates the user and student data currently on DB.
+   * @param studentUserData student user data to be updated.
+   * @param auditUserId user that should be considered the one that is causing the changes.
    * @returns updated student, if some data was changed.
    */
-  async synchronizeFromUserToken(
-    studentToken: StudentUserToken,
-  ): Promise<Student> {
-    const studentToSync = await this.getStudentById(studentToken.studentId);
+  async updateStudentUserData(
+    studentUserData: StudentUserData,
+    auditUserId: number,
+  ): Promise<boolean> {
+    const studentToSync = await this.getStudentById(studentUserData.studentId);
     let mustSave = false;
-    if (studentToken.givenNames === undefined) {
-      studentToken.givenNames = null;
+    if (!studentUserData.givenNames?.trim()) {
+      studentUserData.givenNames = null;
     }
     if (
-      !dayjs(studentToken.birthdate).isSame(studentToSync.birthDate) ||
-      studentToken.lastName !== studentToSync.user.lastName ||
-      studentToken.givenNames !== studentToSync.user.firstName
+      !dayjs(studentUserData.birthdate).isSame(studentToSync.birthDate) ||
+      studentUserData.lastName.toLowerCase() !==
+        studentToSync.user.lastName.toLowerCase() ||
+      studentUserData.givenNames?.toLowerCase() !==
+        studentToSync.user.firstName?.toLowerCase()
     ) {
-      studentToSync.birthDate = studentToken.birthdate;
-      studentToSync.user.lastName = studentToken.lastName;
-      studentToSync.user.firstName = studentToken.givenNames;
+      studentToSync.birthDate = studentUserData.birthdate;
+      studentToSync.user.lastName = studentUserData.lastName;
+      studentToSync.user.firstName = studentUserData.givenNames;
       const sinValidation = new SINValidation();
       sinValidation.student = studentToSync;
       sinValidation.sin = studentToSync.sinValidation.sin;
       studentToSync.sinValidation = sinValidation;
       mustSave = true;
     }
     // This condition is not added above, as email does not trigger SIN validation request.
-    if (studentToken.email !== studentToSync.user.email) {
-      studentToSync.user.email = studentToken.email;
+    if (studentUserData.email !== studentToSync.user.email) {
+      studentToSync.user.email = studentUserData.email;
       mustSave = true;
     }
 
     if (mustSave) {
-      studentToSync.modifier = { id: studentToken.userId } as User;
-      return await this.save(studentToSync);
+      await this.dataSource.transaction(async (transactionalEntityManager) => {
+        await this.noteSharedService.createStudentNote(
```

Comment:
The note creation should be either conditional or have a hard-coded message in case the user is from BCSC.
I would recommend asking businesses to have a hard-coded message.